### PR TITLE
Have Single Select Refinement UI Change Look Smoother

### LINKED ISSUE #526: Default results appear when switching single-select facets
### Overview:

There is a brief flash of the no-filtered results when switching from one single-select facet to another

search

https://react-commerce-ifixit.vercel.app/Parts
https://www.ifixit.com/Parts

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview

Basically a reopening of #617 now that the `instant-search` update is working on vercel.

When we used to swap single select refinements we would sometimes have the no-refinements selection blip on the screen for a half second (see the issue #526 for an example). This pr should solve that problem by updating the full Index UI State at once (including the refinements) and not clearing out the refinements and then setting them which was why we would have a slight blip of wrong results.

## CR
Is there a more efficient way to organize the data for the IndexUIState than what I have?

## QA
Make sure that pages don't have the blip of wrong results when switching single select refinements (note that even on master this blip disappears on switching if youve already switched between the two refinements on a page load. it must be caching the refinement results in the browser). Make sure that refinements are working as intended. Also make sure that performance isn't down the toilet.

Closes #526

### CODE REVIEW

**Comment by danielcliu-ifixit** [comment]
Something is going on weird with our product metafields and thats making this not work rn. Im going to try to solve that issue first

**Review by masonmcelvain — APPROVED** [review]
CR 🔥

File: frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx [hunk 1]
```diff
@@ -106,7 +89,21 @@ const SingleSelectItem = React.memo(function SingleSelectItem({
...
+            setIndexUiState((prevUiState) => {
+               const refinementList = {
+                  ...prevUiState.refinementList,
+                  [attribute]: [item.value],
+               };
+               if (
+                  item.value === prevUiState?.refinementList?.[attribute]?.[0]
+               ) {
+                  delete refinementList[attribute];
+               }
```
Comment (masonmcelvain) [inline frontend/templates/product-list/sections/FilterableProductsSection/RefinementSingleSelect.tsx:101]: This is the only other way I can think of, but I think what you have is fine

```suggestion
               const refinementList = item.value === prevUiState?.refinementList?.[attribute]?.[0]
                  ? {
                        ...prevUiState.refinementList,
                        [attribute]: undefined,
                  } : {
                        ...prevUiState.refinementList,
                        [attribute]: [item.value],
                  };
```

**Comment by k0rvusk0r4x** [comment]
The blip of wrong results is gone.

However, the images are now taking a rather long time to load.

dev_block 😿

**Comment by sterlinghirsh** [comment]
un_dev_block :zap: I am not seeing a blip nor particularly slow loading of images. Given that no image code was changed, I'm going to say let's merge this and if we see poor image performance going forward, we can look into it or revert.

**Comment by sterlinghirsh** [comment]
QA :zap:

**Pull request merged**